### ISSUE TITLE
Problem with separating line - again

### ISSUE THREAD

**Issue by jaclas** [cnpack/cnwizards#58]
![vps_07](https://cloud.githubusercontent.com/assets/17312717/25856181/881409a8-34d5-11e7-8172-32d4e5c6dc11.png)

Delphi Tokyo

**Comment by jaclas** [comment]
Example code:

`unit ss.Cuculus.Types;

interface


implementation

uses
  Spring;

function RAnnouncementConditionsData.CalcConditionsHash(): UInt32;
var
  lCalcString: TAction<string>;
  lStrToHash : TStringBuilder;
begin
  lStrToHash := TStringBuilder.Create;
  try
  lCalcString := procedure(const aStr : string)
                 begin
                   lStrToHash.Append(aStr);
                 end;
  TArray.ForEach<string>(VATIDArray, lCalcString);

  fPKDConditions.ForEach(lCalcString);
  fCommuneConditions.ForEach(lCalcString);
  fDistrictConditions.ForEach(lCalcString);
  fPlaceConditions.ForEach(lCalcString);
  fProductConditions.ForEach(procedure(const aProd : IProductCondition)
                             begin
                              lStrToHash.Append(aProd.ProductId);
                              lStrToHash.Append(aProd.ProductVersion);
                              lStrToHash.Append(aProd.ProductReleaseDate.ToString);
                              lStrToHash.Append(aProd.DateConditionAsString);
                              lStrToHash.Append(aProd.VersionConditionAsString);
                             end);
  lStrToHash.Append(fDemoVersionMode.ToChar);
  lStrToHash.Append(fPersonKind.ToChar);
  lStrToHash.Append(fRunBusinessInDays);
  lStrToHash.Append(fRunBusinessInDaysCondition.ToSymbol);
  Result := THasher.CRC32StringToUInt32(lStrToHash.ToString);
  finally
  lStrToHash.Free;
  end;
end;

end.`



Without this code block:

`  fProductConditions.ForEach(procedure(const aProd : IProductCondition)
                             begin
                              lStrToHash.Append(aProd.ProductId);
                              lStrToHash.Append(aProd.ProductVersion);
                              lStrToHash.Append(aProd.ProductReleaseDate.ToString);
                              lStrToHash.Append(aProd.DateConditionAsString);
                              lStrToHash.Append(aProd.VersionConditionAsString);
                             end);`

separating line dissapear.

**Comment by shanzhashu** [comment]
Anonymous functions brings much mess. Now fixed in 899. Could you download it and verify it?

http://www.cnpack.org/downbuilds.php

**Comment by jaclas** [comment]
Yes, line dissapear but... new version of CNPack broke loading EurekaLog:

![vps_03](https://cloud.githubusercontent.com/assets/17312717/25894371/d184326a-357b-11e7-9108-b979732580f3.png)

**Comment by jaclas** [comment]
Translate from polish message: 

"Błąd wykonania operacji inpage"

to english:

"Error performing inpage operation"

**Comment by jaclas** [comment]
Next issue:

separated lines exists

![vps_2017 09 25_04](https://user-images.githubusercontent.com/17312717/30812477-1948119c-a20b-11e7-8500-7dddbc7ec399.png)

separated lines disappear
![vps_2017 09 25_03](https://user-images.githubusercontent.com/17312717/30812480-1bff468a-a20b-11e7-9398-c4e7963a1b37.png)


test code:

```
unit Unit1;

interface

implementation

procedure TServiceClientConfiguration.StoreCustomValue(const aNodePath, aName: string; const aValue: Int64);
begin
  fConfigurationStorageDriver.SetOnceNodePathTo(aNodePath).StoreValue(aName, aValue);
end;

procedure TServiceClientConfiguration.StoreCustomValue(const aNodePath, aName, aValue: string);
begin
  fConfigurationStorageDriver.SetOnceNodePathTo(aNodePath).StoreValue(aName, aValue);
end;

procedure TServiceClientConfiguration.StoreCustomValue(const aNodePath, aName: string; const aValue: Variant);
begin

end;

function TServiceClientConfiguration.RetoreCustomValue(const aNodePath, aName: string; const aDefaultValue: Boolean): Boolean;
begin

end;

function TServiceClientConfiguration.RetoreCustomValue(const aNodePath,
  aName: string; const aDefaultValue: TDateTime): TDateTimete;
begin

end;

function TServiceClientConfiguration.RetoreCustomValue(const aNodePath,
 aName: string; const aDefaultValue: Int64): Int64;
begin

end;

end.

```

**Comment by shanzhashu** [comment]
Are there any strange characters in the source such as #0? 

I just copy the code to my IDE and seems all are OK.

**Comment by jaclas** [comment]
No,  but for sure I now copied code from this issue and problem still occurs

**Comment by jaclas** [comment]
![vps_2018 03 27_02](https://user-images.githubusercontent.com/17312717/37974996-98e78b68-31de-11e8-99f9-5fde686beef0.png)

![vps_2018 03 27_03](https://user-images.githubusercontent.com/17312717/37975020-a2d70662-31de-11e8-8897-6c5d59d69cfe.png)

code (in delphi file format set to utf8):


```
unit Unit1;

interface

implementation

function CurrentDate(): TDate;
var
  lDate: TDate;
begin
  if (GetTickCount64 - fCDDownCounter) > 0 then
  begin
    lDate := Date();
    if lDate > fCurrentDate then
    begin
      fCurrentDate := lDate;
      fCDDownCounter := GetTickCount64 + Now.MilliSecondsBetween(fCurrentDate + 1) + 1000;
    end;
  end;
  Result := fCurrentDate;
end;


function GetOwner(aIndex: Integer): IBaseInstrumental;
begin
  Result := fOwnerList.Items[aIndex];
end;

function GetOwnerClass(aIndex: Integer): TClass;
begin
  Result := TObject(fOwnerList.Items[aIndex]).ClassType;
end;

function GetResult(const aSQL : string; aParams: array of Variant): IDataMatrix;
begin
  Result := GetInternalResult(aSQL, aParams);
end;

end.

```
